File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/analyticssubmission/model/AnalyticsWindow.java
package uk.nhs.nhsx.analyticssubmission.model;

import uk.nhs.nhsx.core.DateFormatValidator;

public class AnalyticsWindow {
    
    public final String startDate;
    public final String endDate;

    public AnalyticsWindow(String startDate, String endDate) {
        DateFormatValidator
            .toZonedDateTimeMaybe(startDate)
            .orElseThrow(() -> new IllegalArgumentException("Invalid startDate format"));
        DateFormatValidator
            .toZonedDateTimeMaybe(endDate)
            .orElseThrow(() -> new IllegalArgumentException("Invalid endDate format"));
        this.startDate = startDate;
        this.endDate = endDate;
    }
}

File: src/aws/lambdas/incremental_distribution/cta/src/main/java/uk/nhs/nhsx/analyticssubmission/model/AnalyticsWindow.kt
package uk.nhs.nhsx.analyticssubmission.model

import uk.nhs.nhsx.core.Clock
import java.time.Duration
import java.time.Instant

data class AnalyticsWindow(
    val startDate: Instant,
    val endDate: Instant
) {

    fun isDateRangeInvalid(clock: Clock): Boolean {
        val outOfScope = clock().plus(Duration.ofDays(365))
        return startDate.isAfter(outOfScope) || endDate.isAfter(outOfScope)
    }
}
